src/components/Header.tsx

```diff
@@ -13,15 +20,33 @@ const HeaderRow = styled(Row)`
 `
 
 export interface HeaderProps {
-  title?: ReactElement
-  children: ReactNode
+  hideConnectionUI?: boolean
+  onConnectWalletClick?: () => void | Promise<boolean>
 }
 
-export default function Header({ title, children }: HeaderProps) {
+export default function Header(props: HeaderProps) {
+  const { isActive } = useWeb3React()
+  const onSupportedNetwork = useOnSupportedNetwork()
+  const isDisabled = !(isActive && onSupportedNetwork)
+
+  const [onConnectWalletClick, setOnConnectWalletClick] = useAtom(onConnectWalletClickAtom)
+  useEffect(() => {
+    if (props.onConnectWalletClick !== onConnectWalletClick) {
+      setOnConnectWalletClick((old: (() => void | Promise<boolean>) | undefined) => (old = props.onConnectWalletClick))
+    }
+  }, [props.onConnectWalletClick, onConnectWalletClick, setOnConnectWalletClick])
+
   return (
     <HeaderRow iconSize={1.2}>
-      <Row gap={0.5}>{title && <ThemedText.Subhead1>{title}</ThemedText.Subhead1>}</Row>
-      <Row gap={1}>{children}</Row>
+      <Row gap={0.5}>
+        <ThemedText.Subhead1>
+          <Trans>Swap</Trans>
```

Comment:
Should widgets expand past `Swap`, this will need to be configurable.
Can you keep passing `title` into the header?